src/liballoc/arc.rs

```diff
@@ -192,6 +191,45 @@ impl<T> Arc<T> {
         };
         Arc { _ptr: unsafe { NonZero::new(Box::into_raw(x)) } }
     }
+
+    /// Unwraps the contained value if the `Arc<T>` has only one strong reference.
+    /// This will succeed even if there are outstanding weak references.
+    ///
+    /// Otherwise, an `Err` is returned with the same `Arc<T>`.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::sync::Arc;
+    ///
+    /// let x = Arc::new(3);
+    /// assert_eq!(Arc::try_unwrap(x), Ok(3));
+    ///
+    /// let x = Arc::new(4);
+    /// let _y = x.clone();
+    /// assert_eq!(Arc::try_unwrap(x), Err(Arc::new(4)));
+    /// ```
+    #[inline]
+    #[stable(feature = "rc_unique", since="1.4.0")]
+    pub fn try_unwrap(self) -> Result<T, Arc<T>> {
```

Comment:
I meant that it should have the signature `pub fn try_unwrap(this: Arc<T>) -> Result<T, Arc<T>>` rather than `pub fn try_unwrap(self) -> Result<T, Arc<T>>`.